hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -54,6 +54,24 @@ namespace hardware_interface
  * the names of interface types derived from \ref HardwareInterface and
  * instances of those interface types.
  *
+ * A class derived from this base interface represents a robot and stores the
+ * state of the robot's hardware resources (joints, sensors, actuators) and
+ * outgoing commands in its data member arrays. The names of these members
+ * should provide semantic meaning (e.g. pos, vel, eff, cmd).
+ * Commands from controllers are populated by the controller's update method,
+ * (controller_interface::ControllerBase::update()).
+ * For each resource a \ref JointStateHandle (for read only joints),
+ * \ref JointHandle (for read and write joints) or custom handle can be used
```

Comment:
This question is similar to the [previous one](https://github.com/ros-controls/ros_control/pull/433/files#r405166152). Examples I've seen so far use both (`JointStateHandle` and `JointHandle`) but wouldn't `JointHandle` be enough as it can be used for reading joint state and writing commands?

Reply:
`JointStateHandle` doesn't create a conflict between controllers as it is read-only.